<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CommunicationDialog</class>
 <widget class="QDialog" name="CommunicationDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>817</width>
    <height>276</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Communication Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select a device to change its communication settings and test its connection.&lt;br /&gt;The settings you choose will not be saved until you press the test connection button.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1,1,1">
     <item>
      <widget class="QGroupBox" name="gpibBox">
       <property name="title">
        <string>GPIB</string>
       </property>
       <layout class="QFormLayout" name="formLayout_3">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::ExpandingFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="deviceLabel">
          <property name="text">
           <string>Device</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="gpibDeviceComboBox"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="busAddressLabel">
          <property name="text">
           <string>Bus Address</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="busAddressSpinBox">
          <property name="maximum">
           <number>32</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QPushButton" name="gpibTestButton">
          <property name="text">
           <string>Test Connection</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="tcpBox">
       <property name="title">
        <string>TCP</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="deviceLabel_2">
          <property name="text">
           <string>Device</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="tcpDeviceComboBox"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="ipAddressLabel">
          <property name="text">
           <string>IP Address</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="ipLineEdit"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="portLabel_3">
          <property name="text">
           <string>Port</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="portSpinBox">
          <property name="maximum">
           <number>2147483647</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QPushButton" name="tcpTestButton">
          <property name="text">
           <string>Test Connection</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="rs232Box">
       <property name="title">
        <string>RS232</string>
       </property>
       <layout class="QFormLayout" name="formLayout_11">
        <item row="0" column="0">
         <widget class="QLabel" name="deviceLabel_3">
          <property name="text">
           <string>Device</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="rs232DeviceComboBox"/>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QPushButton" name="rs232TestButton">
          <property name="text">
           <string>Test Connection</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="deviceIDLabel">
          <property name="text">
           <string>Device ID</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="rs232DeviceIDLineEdit"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="baudRateLabel_2">
          <property name="text">
           <string>Baud Rate</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="baudRateComboBox">
          <property name="currentIndex">
           <number>-1</number>
          </property>
          <item>
           <property name="text">
            <string>300</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>14400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>28800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>38400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>57600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>115200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>230400</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="customBox">
       <property name="title">
        <string>Custom</string>
       </property>
       <layout class="QFormLayout" name="customBoxLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Device</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="customDeviceComboBox"/>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QPushButton" name="customTestButton">
          <property name="text">
           <string>Test Connection</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>gpibDeviceComboBox</tabstop>
  <tabstop>busAddressSpinBox</tabstop>
  <tabstop>gpibTestButton</tabstop>
  <tabstop>tcpDeviceComboBox</tabstop>
  <tabstop>ipLineEdit</tabstop>
  <tabstop>portSpinBox</tabstop>
  <tabstop>tcpTestButton</tabstop>
  <tabstop>rs232DeviceComboBox</tabstop>
  <tabstop>rs232DeviceIDLineEdit</tabstop>
  <tabstop>baudRateComboBox</tabstop>
  <tabstop>rs232TestButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CommunicationDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CommunicationDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
